### PR TITLE
use current workspace in vscode

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: backend/src/exploregens.ts [hunk 1]
```diff
@@ -99,7 +88,11 @@ export class ExploreGens {
                 await this.updateAllInstalledGenerators();
             }
         }
-    }
+	}
+	
+	private isInTheia(): boolean {
+		return this.inTheia;
+	}
```
Comment (tomer-epstein) [inline backend/src/exploregens.ts:95]: no need for this func. can directly use this.isInTheia member.

File: backend/src/extension.ts [hunk 2]
```diff
@@ -20,19 +21,27 @@ export function activate(context: vscode.ExtensionContext) {
...
 	registerAndSubscribeCommand("exploreGenerators", exploreGensPanel.loadWebviewPanel.bind(exploreGensPanel));
 	registerWebviewPanelSerializer(exploreGensPanel);
 }
 
+async function isInTheia() {
+	const commands = await vscode.commands.getCommands(true);
+	const foundCommands = _.intersection(commands, ["theia.open", "preferences:open", "keymaps:open", "workspace:openRecent"]);
+	return !_.isEmpty(foundCommands);
+}
+
```
Comment (tomer-epstein) [inline backend/src/extension.ts:44]: We had 2 months ago class named Theia.ts
https://github.com/SAP/yeoman-ui/commit/d109bea2bc7c9298683fee9bac5b3895da241851#diff-cf07a938a7f1b68c87573b5872d099b1

File: backend/src/panels/AbstractWebviewPanel.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ export abstract class AbstractWebviewPanel {
 
 	protected logger: IChildLogger;
 	protected disposables: vscode.Disposable[];
+	protected inTheia: boolean;
```
Comment (tomer-epstein) [inline backend/src/panels/AbstractWebviewPanel.ts:23]: protected isInTheia: boolean;

File: backend/src/exploregens.ts [hunk 4]
```diff
@@ -29,10 +29,9 @@ export class ExploreGens {
     private cachedInstalledGeneratorsPromise: Promise<string[]>;
     private readonly context: any;
     private readonly vscode: any;
-    private isInTheiaCached: boolean;
+    private inTheia: boolean;
```
[the remark this answers is not in the document]
Comment (slavik-lvovsky) [reply backend/src/exploregens.ts:32]: There is already method with this name

File: backend/src/exploregens.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply backend/src/exploregens.ts:95]: There is already method with this name

File: backend/src/panels/AbstractWebviewPanel.ts (hunk 3 above)
Comment (slavik-lvovsky) [reply backend/src/panels/AbstractWebviewPanel.ts:23]: There is already method with this name

File: backend/src/extension.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply backend/src/extension.ts:44]: We need to calculate it only once

File: backend/src/exploregens.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply backend/src/exploregens.ts:95]: This method is used in UI through rpc

**Comment by idoprz** [comment]
Why do you need this?

**Comment by slavik-lvovsky** [comment]
> Why do you need this?

I need to calculate default outputPath for generator: in app studio it should be projects, in vscode it may be current opened workspace

File: backend/src/exploregens.ts (hunk 1 above)
Comment (tomer-epstein) [reply backend/src/exploregens.ts:95]: why there's a need for this rpc func. you are passing the boolean isInTheia to exolregens constructor.

File: backend/src/panels/AbstractWebviewPanel.ts (hunk 3 above)
Comment (tomer-epstein) [reply backend/src/panels/AbstractWebviewPanel.ts:23]: youhave set it onAbstractWebviewPanel but didn't use it.

**Comment by idoprz** [comment]
So you should check if you are in AppStudio, not if you are in Theia. We use the env variable WS_BASE_URL which exist only in AppStudio.

File: backend/src/panels/AbstractWebviewPanel.ts (hunk 3 above)
Comment (slavik-lvovsky) [reply backend/src/panels/AbstractWebviewPanel.ts:23]: It is declared once in this abstract panel but used in both other panels: yeomanui panel and explogen panel

Comment (tomer-epstein) [reply backend/src/panels/AbstractWebviewPanel.ts]: I see now. olease use same param names in both panel --> isInTheia: boolean

File: backend/src/exploregens.ts [hunk 5]
```diff
@@ -99,7 +88,11 @@ export class ExploreGens {
                 await this.updateAllInstalledGenerators();
             }
         }
-    }
+	}
+	
+	private getInBAS(): boolean {
```
Comment (tomer-epstein) [inline backend/src/exploregens.ts:93]: private getIsInBAS(): boolean {

File: backend/src/exploregens.ts [hunk 6]
```diff
@@ -108,7 +101,7 @@ export class ExploreGens {
         this.rpc.registerMethod({ func: this.uninstall, thisArg: this });
         this.rpc.registerMethod({ func: this.isInstalled, thisArg: this });
         this.rpc.registerMethod({ func: this.getRecommendedQuery, thisArg: this });
-        this.rpc.registerMethod({ func: this.isInTheia, thisArg: this });
+        this.rpc.registerMethod({ func: this.getInBAS, thisArg: this });
```
Comment (tomer-epstein) [inline backend/src/exploregens.ts:104]: getIsInBAS

File: frontend/src/ExploreGensApp.vue [hunk 7]
```diff
@@ -199,8 +199,8 @@ export default {
       this.items = await this.rpc.invoke("getRecommendedQuery");
       this.items.unshift(ALL_GENS);
     },
-    async setIsInTheia() {
-      this.isInTheia = await this.rpc.invoke("isInTheia");
+    async setIsInBAS() {
+      this.isInBAS = await this.rpc.invoke("getInBAS");
```
Comment (tomer-epstein) [inline frontend/src/ExploreGensApp.vue:203]: getIsInBAS

**Review by tomer-epstein — CHANGES_REQUESTED** [review]

File: backend/src/panels/AbstractWebviewPanel.ts [hunk 8]
```diff
@@ -37,6 +38,7 @@ export abstract class AbstractWebviewPanel {
 		this.logger = getLogger();
 		this.disposables = [];
 		this.context = context;
+		this.isInBAS = !_.isEmpty(_.get(process, "env.WS_BASE_URL"));
```
Comment (tomer-epstein) [inline backend/src/panels/AbstractWebviewPanel.ts:41]: Nice. 💯

File: frontend/src/ExploreGensApp.vue (hunk 7 above)
Comment (slavik-lvovsky) [reply frontend/src/ExploreGensApp.vue:203]: done

Comment (tomer-epstein) [reply frontend/src/ExploreGensApp.vue:203]: async setIsInBAS() {
      this.isInBAS = await this.rpc.invoke("getIsInBAS");

File: frontend/src/ExploreGensApp.vue [hunk 9]
```diff
@@ -262,7 +262,7 @@ export default {
     await this.setupRpc();
     await Promise.all([
       await this.setIsLegalNoteAccepted(),
-      await this.setIsInBAS()
+      await this.getIsInBAS()
```
Comment (tomer-epstein) [inline frontend/src/ExploreGensApp.vue:265]: await this.setIsInBAS()

**Review by tomer-epstein — CHANGES_REQUESTED** [review]

**Review by tomer-epstein — APPROVED** [review]

**Pull request merged**